File: CSharp/base64encoding.cs
using System;
using System.IO;
using System.Net;
using System.Text;

namespace Examples.System.Net
{
    public class WebRequestPostExample
    {
        public static void Main()
        {
            // Your customer key
            string customerKey = "Your customer key";
            // Your customer secret
            string customerSecret = "Your customer secret";

            string plainCredential = customerKey + ":" + customerSecret;

            // Perform base64 encoding
            var plainTextBytes = Encoding.UTF8.GetBytes(plainCredential);
            string encodedCredential = Convert.ToBase64String(plainTextBytes);
            // Create authorization header
            string authorizationHeader = "Authorization: Basic " + encodedCredential + "\"";

            // Create a request using a URL that can receive a post.
            WebRequest request = WebRequest.Create("https://api.agora.io/dev/v2/project/<Your App ID>/rtm/users/userA/peer_messages?wait_for_ack=true ");
            // Set the Method property of the request to POST.
            request.Method = "POST";

            // Create POST data and convert it to a byte array.
            string postData = "{\"destination\": \"userB\",\r\n    \"enable_offline_messaging\": false,\r\n    \"enable_historical_messaging\": false,\r\n    \"payload\": \"Hello\"\r\n}";
            byte[] byteArray = Encoding.UTF8.GetBytes(postData);

            // Set the ContentType property of the WebRequest.
            request.Headers.Add(authorizationHeader);
            request.ContentType = "text/plain";


            // Get the request stream.
            Stream dataStream = request.GetRequestStream();
            // Write the data to the request stream.
            dataStream.Write(byteArray, 0, byteArray.Length);
            // Close the Stream object.
            dataStream.Close();

            // Get the response.
            WebResponse response = request.GetResponse();
            // Display the status.
            Console.WriteLine(((HttpWebResponse)response).StatusDescription);

            // Get the stream containing content returned by the server.
            // The using block ensures the stream is automatically closed.
            using (dataStream = response.GetResponseStream())
            {
                // Open the stream using a StreamReader for easy access.
                StreamReader reader = new StreamReader(dataStream);
                // Read the content.
                string responseFromServer = reader.ReadToEnd();
                // Display the content.
                Console.WriteLine(responseFromServer);
            }

            // Close the response.
            response.Close();
        }
    }
}

File: nodejs/base64encoding.js
// Basic authentication example
const https = require('https')
// Customer key
const customerKey = "Your customer key"
// Customer secret
const customerSecret = "Your customer secret"
// Create plain credential for base64 encoding
const plainCredential = customerKey + ":" + customerSecret
// Perform base64 encoding
encodedCredential = Buffer.from(plainCredential).toString('base64')
authorizationField = "Basic " + encodedCredential

// Set options for the request
const options = {
  hostname: 'api.agora.io',
  port: 443,
  path: '/dev/v2/project/<Your App ID>/rtm/users/userA/peer_messages?wait_for_ack=true',
  method: 'POST',
  headers: {
    'Authorization':authorizationField,
    'Content-Type': 'application/json'
  }
}
// Set body
const data = JSON.stringify({"destination": "userB", "enable_offline_messaging": false,  "enable_historical_messaging": false, "payload": "Hello"})

const req = https.request(options, res => {
  console.log(`Status code: ${res.statusCode}`)

  res.on('data', d => {
    process.stdout.write(d)
  })
})

req.on('error', error => {
  console.error(error)
})

req.write(data)
req.end()